packages/pyright-internal/src/analyzer/checker.ts

```diff
@@ -780,11 +780,13 @@ export class Checker extends ParseTreeWalker {
                             param.d.name
                         );
                     } else if (isAny(paramType)) {
-                        this._evaluator.addDiagnostic(
-                            DiagnosticRule.reportAny,
-                            LocMessage.paramTypeAny().format({ paramName: param.d.name.d.value }),
-                            param.d.name
-                        );
+                        if (isAny(paramType) && this._shouldReportAny(param)) {
+                            this._evaluator.addDiagnostic(
+                                DiagnosticRule.reportAny,
+                                LocMessage.paramTypeAny().format({ paramName: param.d.name.d.value }),
+                                param.d.name
+                            );
+                        }
```

Comment:
this change doesn't seem to address the example in #1235. do you mind providing some more info about the exact issue you encountered that this fixes? the unit tests you added don't seem to report a `reportAny` diagnostic in the main branch so it's difficult to tell what this is changing exactly.

i think to address the example in #1235 this check needs to instead be added here: https://github.com/DetachHead/basedpyright/blob/a8a0d2f9214cfa9b790f92cdcdbbe9d6a0a75ed7/packages/pyright-internal/src/analyzer/typeEvaluator.ts#L15318-L15320